### PR TITLE
gslab-econ/template#88 recursive fun to flatten dicts

### PR DESCRIPTION
**Pull request by arosenbe**
@stanfordquan, can you review this portion of work from gslab-econ/template#88? It's just a recursive function to flatten dictionaries based on [SO](https://stackoverflow.com/a/6027615/7667392).

### CODE REVIEW

File: gslab_scons/misc.py [hunk 1]
```diff
@@ -358,3 +358,15 @@ def finder(rel_parent_dir, pattern, excluded_dirs = []):
...
+def flatten_dict(d, parent_key = '', sep = ':'):
+    items = []
+    for k, v in d.items():
+        new_key = parent_key + sep + k if parent_key else k
+        try:
+            items.extend(flatten_dict(v, new_key, sep = sep).items())
+        except:
+            print new_key
+            items.append((new_key, v))
+    print dict(items)
```
Comment (stanfordquan) [inline gslab_scons/misc.py]: do we want this print statement?

File: gslab_scons/misc.py [hunk 2]
```diff
@@ -358,3 +358,15 @@ def finder(rel_parent_dir, pattern, excluded_dirs = []):
         out_paths = []
 
     return out_paths
+
+def flatten_dict(d, parent_key = '', sep = ':'):
+    items = []
```
Comment (stanfordquan) [inline gslab_scons/misc.py]: please add docstring

**Comment by arosenbe** [comment]
Thanks @stanfordquan. the commit above takes care of your comments. It also adds a new feature to avoid clobbering the values of repeated keys. Could you take a look at that too?

File: gslab_scons/misc.py [hunk 3]
```diff
@@ -359,14 +359,39 @@ def finder(rel_parent_dir, pattern, excluded_dirs = []):
...
-        new_key = parent_key + sep + k if parent_key else k
-        try:
-            items.extend(flatten_dict(v, new_key, sep = sep).items())
-        except:
-            print new_key
-            items.append((new_key, v))
-    print dict(items)
+    for key, val in sorted(d.items()):
+        # Create name of new key
+        new_key = parent_key + sep + key if parent_key else key
```
[the remark this answers is not in the document]
Comment (arosenbe) [reply gslab_scons/misc.py]: Good suggestion. Done.

**Comment by arosenbe** [comment]
PR passed, merging.

**Pull request merged**